torch/distributed/utils.py

```diff
@@ -4,10 +4,40 @@
 from torch.nn.parallel.scatter_gather import (  # type: ignore[attr-defined]
     _is_namedtuple
 )
-from typing import Dict, Any, List
+from typing import Dict, Any, List, Tuple
 
 __all__ = []  # type: ignore[var-annotated]
 
+def _pack_kwargs(*args: Any, **kwargs: Any) -> Tuple[Tuple[str, ...], Tuple[Any, ...]]:
+    """
+    Turn argument list into separate key list and value list (unpack_kwargs does the opposite)
```

Comment:
I realize this is a private function, but could we consider adding a section to the docstring like:
```
Returns:
    Tuple[Tuple[str, ...], Tuple[Any, ...]]: The first tuple element gives
    the kwarg keys, and the second tuple element gives both positional args
    and kwarg values, where the positional args proceed kwarg values and
    kwarg values are ordered consistently with the kwarg keys. The first tuple
    element's length is at most the second tuple element's length.
```
My motivation for this is to document the implicit assumptions that the `_unpack_kwargs` method has on the `_pack_kwargs` return value.

For Python 3.7+, the order of `dict` iteration is guaranteed to be the insertion order, so it may be okay to not specify that the returned `tuple` may not match the _original_ kwargs order (if `kwargs.items()` did not guarantee order). The important thing is that the order between the 1st and 2nd returned `tuple` elements are consistent.

Reply:
Thanks for the suggestion, added this docstring!